src/backend/metal/src/command.rs

```diff
@@ -996,32 +1063,22 @@ impl RawCommandQueue<Backend> for CommandQueue {
         }
     }
 
-    fn present<IS, IW>(&mut self, swapchains: IS, _wait_semaphores: IW) -> Result<(), ()>
+    fn present<IS, S, IW>(&mut self, swapchains: IS, _wait_semaphores: IW) -> Result<(), ()>
     where
-        IS: IntoIterator,
-        IS::Item: borrow::BorrowMut<window::Swapchain>,
+        IS: IntoIterator<Item = (S, FrameImage)>,
+        S: Borrow<window::Swapchain>,
         IW: IntoIterator,
         IW::Item: Borrow<native::Semaphore>,
     {
-        use std::borrow::BorrowMut;
-        for mut swapchain in swapchains {
+        for (swapchain, index) in swapchains {
             // TODO: wait for semaphores
-            let swapchain = swapchain.borrow_mut();
-            let (surface, io_surface, present_id) = swapchain.present();
-            debug!("presenting frame {}", present_id);
-            unsafe {
-                let render_layer_borrow = surface.render_layer.lock().unwrap();
-                let render_layer = *render_layer_borrow;
-                msg_send![render_layer, setContents: io_surface.obj];
-            }
+            debug!("presenting frame {}", index);
+            swapchain.borrow().present(index);
         }
 
-        if cfg!(feature = "auto-capture") {
-            let shared_capture_manager = CaptureManager::shared();
-            if let Some(default_capture_scope) = shared_capture_manager.default_capture_scope() {
-                default_capture_scope.end_scope();
-                default_capture_scope.begin_scope();
-            }
+        let shared_capture_manager = CaptureManager::shared();
```

Comment:
Why remove the feature here?

Reply:
Because we want to stop the capture here unconditionally. The feature allows us to capture workloads that don't involve presenting a frame, e.g. compute/transfers/tests. Here though in most cases I could think of stopping the capture is the right solution.